search/src/Components/upload/react-images-upload/index.js

```diff
@@ -77,6 +78,7 @@ class ReactImageUploadComponent extends React.Component {
           type: ERROR.NOT_SUPPORTED_EXTENSION
         });
         fileErrors.push(fileError);
+        this.setState({fileErrors});
```

Comment:
Why do we need this here? Isn't setting the errors after creating all the promises fine?

Reply:
This triggers to show error of each image. I think it's the most proper way as you don't want to inform users the format is wrong after the whole dataset being uploaded.

Reply:
I think the logic in the existing library has some sense to it: files that match the expected format are uploaded, while those that do not match are not uploaded. Also, given the asynchronous image loading and uploading, the user will be informed at roughly the same time since queuing all the images to load is fairly cheap.
However, your change triggers multiple successive renders, when each error is added, so is not great for responsiveness.